### ISSUE TITLE
Phase 2: Bayesian optimizer (GP/BO) under the LLM proposal engine

### ISSUE THREAD

**Issue by DavidJBarnes** [DavidJBarnes/Formulation-AI#12]
## Background

Phase 1 uses Claude Opus 4.7 as a pure LLM proposal engine — it reads base products + tested results and proposes the next candidates with predicted values ± 1σ. This works for demos but has known weaknesses at scale: cost, latency, and no mathematical convergence guarantee.

Phase 2 replaces the LLM's internal guessing with a real Gaussian Process / Bayesian Optimization loop, while keeping the LLM as the **reasoning and rationale layer** on top of the math.

## Architecture contract (already in place)

`services/proposal_engine.py` is the only module the router calls. The output contract it must satisfy — unchanged between Phase 1 and 2:

\`\`\`python
@dataclass
class ProposedFormulation:
    label: str
    rationale: str                     # LLM-generated explanation
    ingredients: dict[str, float]      # name → amount
    predictions: list[dict]            # [{property, value, sigma}]
\`\`\`

The router, DB layer, and frontend are **blind to whether predictions come from an LLM or a GP** — no changes needed outside `optimizer/`.

## Proposed module structure

\`\`\`
backend/src/formulation_ai/
├── services/
│   └── proposal_engine.py       ← dispatcher: GP when data ≥ threshold, else LLM-only
└── optimizer/
    ├── __init__.py
    ├── gp_model.py              ← GP surrogate (scikit-learn first, BoTorch later)
    ├── acquisition.py           ← EI / UCB / PI acquisition functions
    ├── candidate_selector.py    ← optimize acquisition → top-K candidates
    └── scaler.py                ← ingredient + property normalization
\`\`\`

### Global settings (`config.py`, `FA_` prefix)

| Setting | Type | Default | Purpose |
|---|---|---|---|
| `optimizer_backend` | `"llm" \| "gp_sklearn" \| "gp_botorch"` | `"llm"` | Phase 1→2 switch without code deploy |
| `optimizer_min_observations` | `int` | `3` | Tested rows needed before GP activates; below this falls back to LLM-only |
| `optimizer_acquisition` | `"ei" \| "ucb" \| "pi"` | `"ei"` | Acquisition function |
| `optimizer_ucb_kappa` | `float` | `2.0` | UCB exploration weight (higher = more explore) |
| `optimizer_ei_xi` | `float` | `0.01` | EI jitter (higher = more explore) |
| `optimizer_n_candidates` | `int` | `3` | Default proposals per iteration |
| `optimizer_n_restarts` | `int` | `5` | GP hyperparameter fitting restarts |
| `optimizer_random_seed` | `int \| None` | `None` | None = non-deterministic; set for reproducible demos |

### 1. Scalarization is semantically broken

The main GP trains on `Σ(raw_value × weight)` — a weighted sum of incommensurable raw measurements. For paint, this means maximizing `105×w₁ + 96×w₂ + 4.5×w₃ + 10.5×w₄`, where 105 KU and 4.5 MPa carry wildly different magnitudes. The GP learns to chase large-scale properties and effectively ignores small-magnitude ones.

**Fix:** Normalize each property to its goal before scalarizing. Examples: `score(Tensile ≥55) = value/55`, `score(Viscosity=105) = 1 - |value-105|/105`, `score(Foam ∈ [100,180]) = 1.0 if in-range else penalize by distance`. All four properties land on the same [0,∞) scale; the weighted sum is meaningful.

### 2. Scaler fit on training data instead of design space

`Scaler.fit(X)` uses `X.min/max` across the 3–N tested points. Early iterations, those points cover only a tiny slice of the valid ingredient space. The "bounds" conversion logic that maps declared min/max into scaled space is therefore anchored to the wrong reference frame — the optimizer can wander outside declared bounds or be artificially constrained within them. The `batch_total_g` constraint calculation has the same bug.

**Fix:** Fit the x-scaler on the declared ingredient bounds (`[lo_arr, hi_arr]` as a 2-row matrix), making the declared bounds map exactly to [0, 1]. Scaled bounds become trivially [(0,1), …]. Fix the batch sum constraint to use the correct linear form in scaled space: `x_scaled @ span = batch_total - sum(lo)`.

### 3. Candidates cluster at the same acquisition peak

Running SLSQP from random starts with a 0.05 deduplication threshold still produces three candidates that converge to the global EI maximum — users test nearly identical formulations and learn almost nothing new from the batch.

**Fix:** Kriging Believer — select candidates one at a time: after choosing C1, add `(C1_scaled, GP_predicted_mean)` as a phantom observation to the training set, refit the GP, then optimize for C2 on the updated landscape. The updated GP has lower uncertainty near C1, pushing C2 to a different region. Repeat for C3. Candidates spread naturally without any ad-hoc repulsion penalty.

## Acceptance criteria

- [x] Add optimizer settings to `config.py`
- [x] Add `weight` to `project_targets`, `batch_total_g` to `projects` (migration)
- [x] `optimizer/` module with GP surrogate, EI/UCB/PI acquisition, candidate selector, scaler
- [x] `proposal_engine.py` dispatches to GP when `len(tested) >= optimizer_min_observations`, LLM-only otherwise
- [x] GP predictions feed the LLM rationale prompt (LLM explains GP choices, doesn't guess values)
- [x] Predicted values + 1σ come from GP posterior, not LLM
- [x] Ingredient bounds (`min_amount`/`max_amount`) and `batch_total_g` sum constraint respected
- [x] Unit tests: GP fit, acquisition functions, fallback threshold, constraint enforcement
- [x] `optimizer_backend = "llm"` default keeps Phase 1 behaviour unchanged
- [ ] **Fix #1:** Goal-normalized scalarization (`_property_score`)
- [ ] **Fix #2:** Scaler fit on declared ingredient bounds, not training data; correct linear batch constraint
- [ ] **Fix #3:** Kriging Believer candidate selection for diverse proposals

## Out of scope

- Multi-output BO / Pareto front (separate ticket)
- UI for scalarization weights or sum constraint
- Signals adapter integration
